runtime/interpreter/interpreter.go

```diff
@@ -2214,6 +2217,190 @@ func (interpreter *Interpreter) writeStored(
 	interpreter.recordStorageMutation()
 }
 
+type fromStringFunctionValue struct {
+	receiverType sema.Type
+	hostFunction *HostFunctionValue
+}
+
+// a function that attempts to create a Cadence value from a string, e.g. parsing a number from a string
+type valueParser func(*Interpreter, string) OptionalValue
+
+func newFromStringFunction(ty sema.Type, parser valueParser) fromStringFunctionValue {
+	functionType := sema.FromStringFunctionType(ty)
+
+	hostFunctionImpl := NewUnmeteredHostFunctionValue(
+		func(invocation Invocation) Value {
+			argument, ok := invocation.Arguments[0].(*StringValue)
+			if !ok {
+				// expect typechecker to catch a mismatch here
+				panic(errors.NewUnreachableError())
+			}
+			inter := invocation.Interpreter
+			return parser(inter, argument.Str)
+		},
+		functionType,
+	)
+	return fromStringFunctionValue{
+		receiverType: ty,
+		hostFunction: hostFunctionImpl,
+	}
+}
+
+// default implementation for parsing a given unsigned numeric type from a string.
+// the size provided by sizeInBytes is passed to strconv.ParseUint, ensuring that the parsed value fits in the target type.
+// input strings must not begin with a '+' or '-'.
+func unsignedIntValueParser(bitSize int, convert func(uint64) Value) valueParser {
+	expectedUsage := common.NewNumberMemoryUsage(bitSize / 8)
+
+	return func(interpreter *Interpreter, input string) OptionalValue {
+		val, err := strconv.ParseUint(input, 10, bitSize)
+		if err != nil {
+			return NewNilValue(interpreter)
+		}
+
+		converted := convert(val)
+		common.UseMemory(interpreter, expectedUsage)
+		return NewSomeValueNonCopying(interpreter, converted)
+	}
+}
+
+// default implementation for parsing a given signed numeric type from a string.
+// the size provided by sizeInBytes is passed to strconv.ParseUint, ensuring that the parsed value fits in the target type.
+// input strings may begin with a '+' or '-'.
+func signedIntValueParser(bitSize int, convert func(int64) Value) valueParser {
+	expectedUsage := common.NewCadenceBigIntMemoryUsage(bitSize / 8)
```

Comment:
`NewCadenceBigIntMemoryUsage` (all metering methods/values with `Cadence` prefix) are used for Cadence external values (interpreter values uses a similar set counterparts, without that prefix)

-- sort of exactly why using the 'metered' constructor is safe 😉